head/src/test/kotlin/io/qalipsis/core/head/campaign/states/FactoryAssignmentStateTest.kt

```diff
@@ -278,7 +281,109 @@ internal class FactoryAssignmentStateTest : AbstractStateTest() {
             prop("isSuccessful").isSameAs(false)
         }
         coVerifyOrder {
-            factoryService.checkFactoriesHealth(refEq("my-tenant"), setOf())
+            factoryService.getFactoriesHealth(refEq("my-tenant"), setOf())
+        }
+        confirmVerified(factoryService, campaignReportStateKeeper)
+    }
+
+    @Test
+    fun `should return an aborting state`() = testDispatcherProvider.runTest {
+        // given
+        val state = FactoryAssignmentState(campaign, mockk(), mockk())
```

Comment:
```suggestion
       every { campaign.factories.keys } returns mutableSetOf("node-1", "node-2", "node-3")
        coEvery { factoryService.getFactoriesHealth("my-tenant", mutableSetOf("node-1", "node-2", "node-3")) }      returns emptySet()
        val state = FactoryAssignmentState(campaign, mockk(), mockk())
```

Apply this change in all the test classes.

Reply:
Do not rename the function, I had not seen that you had create the use case as well for the partially unhealthy factories. Just add the mocks in that function.

Reply:
Sorry, the mocks should actually be in the function `should return a disabled state...`